adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m

```diff
@@ -284,38 +301,33 @@ - (nullable NSError *)loadAd:(nonnull NSString *)placement
     NSError *error = nil;
     if (delegate) {
       error = GADMAdapterVungleErrorWithCodeAndDescription(
-                                                           kGADErrorMediationAdapterError, @"A banner ad type has already been "
-                                                           @"instantiated. Multiple banner ads are not "
-                                                           @"supported with Vungle iOS SDK.");
+          kGADErrorMediationAdapterError, @"A banner ad type has already been "
+                                          @"instantiated. Multiple banner ads are not "
+                                          @"supported with Vungle iOS SDK.");
     } else {
       error = GADMAdapterVungleErrorWithCodeAndDescription(
-                                                           kGADErrorMediationAdapterError, @"Can't load ad when try to add a nil delegate.");
+          kGADErrorMediationAdapterError, @"Can't load ad when try to add a nil delegate.");
     }
     return error;
   }
 
   VungleSDK *sdk = [VungleSDK sharedSDK];
   if ([self isAdCachedForPlacementID:placement withDelegate:delegate]) {
     [delegate adAvailable];
-  } else {
-    NSError *loadError;
-    GADMAdapterVungleAdType adType = [delegate adapterAdType];
-    if (adType != GADMAdapterVungleAdTypeBanner && adType != GADMAdapterVungleAdTypeShortBanner && adType != GADMAdapterVungleAdTypeLeaderboardBanner) {
-        if (![sdk loadPlacementWithID:placement error:&loadError]) {
-              if (loadError) {
-                  return loadError;
-              }
-          }
-    } else {
-        if (![sdk loadPlacementWithID:placement withSize:[self getVungleBannerAdSizeType:adType] error:&loadError]) {
-            if ((loadError) && (loadError.code != VungleSDKResetPlacementForDifferentAdSize)) {
-                return loadError;
-            }
-        }
-    }
+    return nil;
   }
 
-  return nil;
+  NSError *loadError = nil;
+  GADMAdapterVungleAdType adType = [delegate adapterAdType];
+  if (adType != GADMAdapterVungleAdTypeBanner && adType != GADMAdapterVungleAdTypeShortBanner &&
+      adType != GADMAdapterVungleAdTypeLeaderboardBanner) {
+    [sdk loadPlacementWithID:placement error:&loadError];
+  } else {
+    [sdk loadPlacementWithID:placement
+                    withSize:[self getVungleBannerAdSizeType:adType]
+                       error:&loadError];
+  }
+  return loadError;
```

Comment:
@ashinagawa Hi Aki, as we discussed in the email, if a placement has banner ad (320x50) cached, but SDK gets `loadPlacementWithID` call with different banner size, Vungle SDK returns "NO" and "non-nil error" for `loadPlacementWithID` call , BUT,  SDK keeps requesting a new ad for this placement, then sending "playabilityUpdate" callback. So we should keep original implementations, `if (loadError.code == VungleSDKResetPlacementForDifferentAdSize)) `, we should not return this loadError, right?  Thanks a lot. ^_^

Reply:
@YueVungle I just added a check for this above, to now revert to the old behavior. I wrote it in such a way that we can easily strip it out once Vungle SDK no longer returns this error.

Please take another look.

Sorry in advance for this still looking like 1 commit. We use [copybara](https://github.com/google/copybara) internally for this, and the tool replaces the commit when I send out for review.

Reply:
@ericleich Hi Eric, thanks for the update, it looks great. 👍O(∩_∩)O